frontend/.claude/skills/frontend-developer/SKILL.md

```diff
@@ -0,0 +1,304 @@
+- Use registry components from `src/components/redpanda-ui/`
+- Install components via CLI: `yes | bunx @fumadocs/cli add --dir https://redpanda-ui-registry.netlify.app/r <component>`
+- Use functional React components with TypeScript
+- Run `bun i --yarn` after installing packages
+
+**NEVER:**
+
+- Use deprecated `@redpanda-data/ui` or Chakra UI
+- Modify files in registry base directory (check `cli.json`)
+- Copy/paste registry source code
+- Add margin `className` directly to registry components
+- Use inline `style` prop on registry components
+- Leave `console.log` or comments in code
+
+## Workflow
+
+### 1. Fetch Registry Documentation
+
+```bash
+# REQUIRED FIRST STEP
+Invoke: mcp__context7__get-library-docs
+Library: /websites/redpanda-ui-registry_netlify_app
+```
+
+### 2. Install Components
+
+```bash
+# Single component
+yes | bunx @fumadocs/cli add --dir https://redpanda-ui-registry.netlify.app/r button
+
+# Multiple components (space-separated)
+yes | bunx @fumadocs/cli add --dir https://redpanda-ui-registry.netlify.app/r card dialog form
+
+# Then generate yarn.lock
+bun i --yarn
+```
+
+### 3. Write Component
+
+**Structure:**
+
+```typescript
+// Functional component with explicit types
+interface Props {
+  title: string;
+  onSubmit: (data: FormData) => void;
+}
+
+export function MyComponent({ title, onSubmit }: Props) {
+  // Component logic
+}
+```
+
+**Styling:**
+
+```typescript
+// ✅ CORRECT: Use component variants
+<Button variant="primary" size="lg">Click</Button>
+
+// ✅ CORRECT: Wrap for spacing
+<div className="mt-4">
+  <Button variant="primary">Click</Button>
+</div>
+
+// ❌ WRONG: Direct margin on component
+<Button className="mt-4" variant="primary">Click</Button>
+```
+
+**TypeScript:**
+
+- Define prop interfaces
+- Never use `any` - infer correct types
+- Use generics for collections
+
+**Performance:**
+
+- Hoist static content outside component
+- Use `useMemo` for expensive computations
+- Use `memo` for components receiving props
+
+### 4. Write Tests
+
+**Test types:**
+
+- Unit tests (`.test.ts`): Pure logic, utilities, helpers - Node environment
+- Integration tests (`.test.tsx`): React components, UI interactions - JSDOM environment
+
+**Unit test example:**
+
+```typescript
+// utils.test.ts
+import { formatNumber } from "./utils";
+
+describe("formatNumber", () => {
+  test("should format numbers with commas", () => {
+    expect(formatNumber(1000)).toBe("1,000");
+  });
+});
+```
+
+**Integration test example:**
+
+```typescript
+// component.test.tsx
+import { render, screen, fireEvent } from 'test-utils';
+import { MyComponent } from './component';
+
+describe('MyComponent', () => {
+  test('should call onSubmit when button clicked', () => {
+    const onSubmit = vi.fn();
+    render(<MyComponent onSubmit={onSubmit} />);
+
+    fireEvent.click(screen.getByRole('button'));
+
+    expect(onSubmit).toHaveBeenCalled();
+  });
+});
+```
```

Comment:
I would replace it with grpc router usage, checking if clicking a button to submit a form would trigger the mutation for example.